### PR TITLE
Use string values in DateFilter null-management constants

### PR DESCRIPTION
**Pull request by teohhanhui**

### CODE REVIEW

**Comment by teohhanhui** [comment]
Do we also want to export these class constants as container parameters?

So that instead of writing

```yaml
    resource.book.date_filter:
        parent: api.doctrine.orm.date_filter
        arguments:
          - datePublished: 0 # Dunglas\ApiBundle\Doctrine\Orm\Filter\DateFilter::EXCLUDE_NULL
```

I can just write

```yaml
    resource.book.date_filter:
        parent: api.doctrine.orm.date_filter
        arguments:
          - datePublished: api.doctrine.orm.date_filter.exclude_null
```

**Comment by sroze** [comment]
IMO it's better to have string values in `DateFilter` constants.

**Comment by ogizanagi** [comment]
:+1: for string values in `DateFilter` constants

**Comment by teohhanhui** [comment]
Are we good?

**Comment by sroze** [comment]
I'm okay with that. But as there's a BC break, I'm waiting another maintainer's :+1:.

File: Resources/doc/filters.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ For instance, exclude entries with a property value of `null`, with the followin
 services:
     resource.date_filter:
         parent:    "api.doctrine.orm.date_filter"
-        arguments: [ { "dateProperty": ~ } ]
```
Comment (theofidry) [inline Resources/doc/filters.md]: Why not letting the default value settable with `~`?

**Comment by dunglas** [comment]
:+1: it's the good time to break things (before 1.0).
Can you also update the changelog please?

File: Resources/doc/filters.md (hunk 1 above)
Comment (dunglas) [inline Resources/doc/filters.md]: Ok with Theo.

Comment (teohhanhui) [inline Resources/doc/filters.md]: The null value is for using database default (i.e. no special null handling). That has not been changed.

What I'm doing here is correcting the example for excluding null values (view the full file to see the text above it).

Comment (theofidry) [inline Resources/doc/filters.md]: Makes sense

**Comment by teohhanhui** [comment]
Added a line in changelog

**Comment by sroze** [comment]
Thank you @teohhanhui ! :+1:

**Pull request merged**